src/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanner.java

```diff
@@ -407,7 +407,9 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 								lifetimeFound = true;
 								lifetimesFound++;
 								//get the portion of the string after "maxage="
-								lifetime = Long.parseLong(directiveToken.substring("max-age=".length()));
+								// Split on comma and use 0th item in case it's malformed: 
+								// Cache-Control: max-age=7776000,private
```

Comment:
Note that this is not malformed, the whitespaces are optional around the comma.

Reply:
Thanks will tweak the wording.